README.md

```diff
@@ -55,6 +55,86 @@ test.all(common)
 
 A valid (read: that follows this interface) IPFS core implementation, must expose the following API.
 
+## Files
+
+### `add`
+
+> Add files and data to IPFS.
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.files.add(data, [callback])
+
+Where `data` may be
+
+- an array of objects, each of the form
+```js
+{
+  path: '/tmp/myfile.txt',
+  content: (Buffer or Readable stream)
+}
+```
+- a `Buffer` instance
+- a `Readable` stream
+
+`callback` must follow `function (err, res) {}` signature, where `err` is an
+error if the operation was not successful. `res` will be an array of
+[DAGNode][]s.
+
+If no `callback` is passed, a promise is returned.
+
+```js
+var files = [
+  {
+    path: '/tmp/myfile.txt',
+    content: fs.createReadStream('/tmp/myfile.txt')
+  }
+]
+ipfs.files.add(files, function (err, dagNodes) {
+  // 'res' will be an array of DAGNodes
+})
+```
+
+
+### `createAddStream`
+
+> Add files and data to IPFS using a transform stream.
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.files.createAddStream([callback])
+
+Provides a Transform stream, where objects can be written of the forms
```

Comment:
Transform streams actually very commonly emit more or fewer elements
than they are given: a zlib compress transform will yield fewer or more
bytes depending on direction; a [binary fsk
encoder](https://github.com/noffle/binary-fsk) will accept byte buffer
objects but emit *way* more audio data depending on the sample rate.

The key distinction is that in Transform streams, the output is in some
way calculated from the input. This is in contrast to a Duplex stream
(e.g. TCP), which is much more akin to two people on a telephone
exchanging messages.